### PR TITLE
Adding escaping of double quotes and tests #3296

### PR DESCRIPTION
**Pull request by jozemlakar**
I have added the escaping and two tests (simple string value and a map)

### CODE REVIEW

File: gatling-core/src/main/scala/io/gatling/core/json/Json.scala [hunk 1]
```diff
@@ -45,7 +45,7 @@ object Json {
   }
 
   private def writeString(s: String, rootLevel: Boolean) = {
-    val escapedLineFeeds = s.replace("\n", "\\n")
+    val escapedLineFeeds = s.replace("\n", "\\n").replace("\"", "\\\"")
```
Comment (slandelle) [inline gatling-core/src/main/scala/io/gatling/core/json/Json.scala]: I don't think that's sufficient. What if content is already escaped?

**Comment by jozemlakar** [comment]
I agree. Should we implement a full escaping system or find an existing one? This fix is for this specific problem and it works. The question is... is it enough? What if \n was already escaped?

**Comment by slandelle** [comment]
> Should we implement a full escaping system or find an existing one?
What if \n was already escaped?

The main problem is we can't know if content is already escaped or not. And I agree there's the same issue if `\n` was already escaped. I'm not sure we'll find a real escaping system that would be smart enough to detect pre-escaped content. We have the same level of complexity for url escaping in gatling-http/AsyncHttpClient.

A real escaping engine that would comply with the full spec would have to support unicode and other weird things. It's definitively beyond the scope here.

I'd be in favor of fix the possible double escaping of double quotes and line feeds and period.

**Comment by jozemlakar** [comment]
I hate to reinvent the wheel so I went and checked what is already available in Gatling. I am familiar with Jackson so I went and used that and it works. Is this the way to go? Should we allow the use of Boon like in parser?

**Comment by slandelle** [comment]
Go with Jackson, Boon is a bit stalled atm.

**Pull request closed without merging**